src/main/java/org/folio/list/services/UserFriendlyQueryService.java

```diff
@@ -112,36 +113,52 @@ private String handleEmpty(EmptyCondition emptyCondition) {
   private <T> String handleConditionWithPossibleIdValue(FieldCondition<T> condition,
                                                         EntityType entityType,
                                                         String userFriendlyOperator,
-                                                        BiFunction<EntityTypeColumn, T, String> labelFn) {
+                                                        BiFunction<Field, T, String> labelFn) {
     String operatorWithPadding = " " + userFriendlyOperator + " ";
     try {
-      return entityType.getColumns()
-        .stream()
-        .filter(column -> column.getSource() != null && condition.field().getColumnName().equals(column.getIdColumnName()))
-        .findFirst()
-        .map(column -> column.getName() + operatorWithPadding + labelFn.apply(column, condition.value()))
-        .orElse(condition.field().getColumnName() + operatorWithPadding + condition.value());
+      // the field referenced directly in the query. May or may not have a source or ID column
+      // e.g. user patron group, vendor code, vendor ID, etc
+      Field querySubjectField = FqlValidationService.findFieldDefinition(condition.field(), entityType).orElseThrow();
+      if (querySubjectField.getIdColumnName() != null) {
+        return condition.field().serialize() + operatorWithPadding + labelFn.apply(querySubjectField, condition.value());
+      } else {
+        // a column which uses our query subject as ID
+        // this will be found when querySubjectField is something like vendor_id, etc.; we would find something like vendor_code
+        Optional<Field> referencingField = entityType.getColumns()
+          .stream()
+          // finds the column referencing this
+          .filter(column -> column.getSource() != null && condition.field().getColumnName().equals(column.getIdColumnName()))
+          .findFirst()
+          .map(Field.class::cast);
```

Comment:
Is this cast necessary? I thought `entityType.getColumns()` already provided `Field`s

Reply:
an `EntityTypeColumn` is a `Field`, but `Optional<EntityTypeColumn>` isn't assignable to `Optional<Field>`.

I can either use `EntityTypeColumn` for the variable declaration, or have this cast. IMO it was best to have the cast since `Field` was used everywhere else